GDALTermProgress: display estimated remaining time for tasks >= 10 seconds

rouault opened the pull request:
Refreshed every tick (so 2.5%)

(only on interactive terminals. Tested with bash on Linux and cmd on Windows, but should work for all reasonable terminals)

```
Before 5 seconds:
0...10...20...

After 5 seconds:
0...10...20...30...40...50                           - estimated remaining time: 00:00:07

Upon completion:
0...10...20...30...40...50...60...70...80...90...100 - done in 00:00:13.
```

Fixes #11100

dbaston commented:
Some changes suggested at https://github.com/rouault/gdal/pull/62

This changes output for me somewhat, e.g. `gdalwarp` in `master` would print

```
Processing /home/dan/Downloads/adaptor.mars.internal-1724252977.7562788-9817-17-5c4869a4-66ce-451e-a86e-66afc72a3873.nc [1/1] : 0..
```
but when the time estimate kicks in, the first part of the line (`Processing...[1/1] : `) is overwritten.

rouault commented on port/cpl_progress.cpp:
```diff
@@ -14,9 +14,16 @@
 
 #include <algorithm>
 
+#ifdef _WIN32
```
done

rouault commented:
> but when the time estimate kicks in, the first part of the line (`Processing...[1/1] : `) is overwritten.

fixed. I've changed the strategy to output as many ``\b`` as emitted during last call
